### ISSUE TITLE
BusinessBase doesn't support string array/list property type.

### ISSUE THREAD

**Issue by realzhaorong** [MarimerLLC/cslaforum#534]
WHY?

**Comment by SachinPNikam** [comment]
List is not Serialize .covert that to byte[] array and send it to server and there you can serialize to original one. using json serialization

**Comment by rockfordlhotka** [comment]
This is because all types in your object graph (so the root object, and then all objects it references) must be serializable. In .NET Core this means they need to implement `IMobileObject`.

The easiest way to do this is to use the [Csla.Core.MobileList](https://github.com/MarimerLLC/csla/blob/master/Source/Csla.Shared/Core/MobileList.cs) type.

**Comment by realzhaorong** [comment]
Can I define a List type like SerializableCollection<T> and marked as Serializable, DONOT implement IMobileObject?

**Comment by rockfordlhotka** [comment]
This depends on the underlying .NET platform.

Full .NET has the BinaryFormatter and NDCS, both of which do 'deep serialization' on any type marked as `Serializable`.

Other .NET platforms don't necessarily have BinaryFormatter. or didn't until recently. For example, there's an outstanding issue for CSLA to allow the use of this serializer in .NET Core: https://github.com/MarimerLLC/csla/issues/703

There's another issue dealing with BinaryFormatter and a bug in full .NET and netstandard that Microsoft will hopefully get resolved soon, as it is a PITA for us at the moment: https://github.com/MarimerLLC/csla/issues/822

**tl;dr** - if you are on a platform that uses `MobileFormatter` you must use types that implement `IMobileObject` or subclass a type where we already implemented it. **Or** you can implement your own wrapping code to otherwise serialize those non-CSLA types in your parent object (assuming the parent object is a CSLA type) by overriding the `IMobileObject` get/set state methods, like we do in the `MobileList` type.

**Comment by Chicagoan2016** [comment]
@rockfordlhotka ,
do you think something like this will work?


`   public static readonly PropertyInfo<MobileList<string>> ListofNamesProperty = RegisterProperty<MobileList<string>>(c => c.ListofNames);
        public MobileList<string> ListofNames
        {
            get { return GetProperty(ListofNamesProperty); }
            set
            {
                SetProperty(ListofNamesProperty, value);
            }
        }`
